### PR TITLE
App Submission: Alby Nostr Wallet Connect

### PR DESCRIPTION
**Pull request by kiwiidb**
This PR adds an application to connect your LND node over Nostr as described in [NIP-47](https://github.com/nostr-protocol/nips/blob/master/47.md).

### CODE REVIEW

**Comment by nmfretz** [comment]
Thank you for this submission @kiwiidb! Super excited to get this on the app store. 

Sorry for the delay in getting to this. I have been a busy 🐝. I will review this week.

File: alby-nostr-wallet-connect/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+services:
+  app_proxy:
+    environment:
+      APP_HOST: alby-nostr-wallet-connect_server_1
+      APP_PORT: 8080
+      PROXY_AUTH_ADD: "false"
+  server:
+    image: ghcr.io/getalby/nostr-wallet-connect:0.1.0@sha256:7e3b59af4abdcbee5def7648ba5253c68e638647bc4a407974a61d13aa2b3b3d
+    volumes:
+      - ${APP_DATA_DIR}/data:/data
```
Comment (nmfretz) [inline alby-nostr-wallet-connect/docker-compose.yml:11]: We need to add the '${APP_DATA_DIR}/data' directory to this repository so that it exists on the user's host machine before Docker Compose is run. When Docker Compose is executed, if the '${APP_DATA_DIR}/data' directory does not exist, Docker will automatically create it as root, while we want to have it owned by the umbrel user (1000). 

Check out [Tailscale](https://github.com/getumbrel/umbrel-apps/tree/master/tailscale) as a direct example of what you can implement. Just create a folder called `data` inside `alby-nostr-wallet-connect` and add an empty `.gitkeep` file to ensure that the directory is tracked by Git

Do you know if the NWC app itself needs to run as root? I see that the database is owned by root:

```
$ ls -l ~/umbrel/app-data/alby-nostr-wallet-connect/data
total 72
-rw-r--r-- 1 root root 69632 Jun 23 05:24 nostr-wallet-connect.db
```

Ideally we would have this running as user 1000:1000.

File: alby-nostr-wallet-connect/umbrel-app.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+manifestVersion: 1
+id: alby-nostr-wallet-connect
+name: Nostr Wallet Connect
+tagline: The power of the zaps at the tip of your fingers.
+icon: https://raw.githubusercontent.com/getAlby/umbrel-community-app-store/master/alby-nostr-wallet-connect/images/nwc_icon.svg
+category: Finance
```
Comment (nmfretz) [inline alby-nostr-wallet-connect/umbrel-app.yml:6]: we just updated our categories yesterday. can you please change to all lowercase `finance`.

```suggestion
category: finance
```

File: alby-nostr-wallet-connect/umbrel-app.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+submitter: kiwiidb
+submission: https://github.com/getAlby/nostr-wallet-connect
+repo: https://github.com/getAlby/nostr-wallet-connect
+support: https://github.com/getAlby/nostr-wallet-connect/issues
+gallery:
+  - https://raw.githubusercontent.com/getAlby/umbrel-community-app-store/master/alby-nostr-wallet-connect/images/nwc_1.png
+  - https://raw.githubusercontent.com/getAlby/umbrel-community-app-store/master/alby-nostr-wallet-connect/images/nwc_2.png
+  - https://raw.githubusercontent.com/getAlby/umbrel-community-app-store/master/alby-nostr-wallet-connect/images/nwc_3.png
+releaseNotes: >-
+  - Budget & time-limited session support
```
Comment (nmfretz) [inline alby-nostr-wallet-connect/umbrel-app.yml:28]: We can leave this blank for the initial Umbrel release. It's primarily used to show users any relevant changes when they get the update notification in their dashboard.

```suggestion
releaseNotes: ""
```

File: alby-nostr-wallet-connect/umbrel-app.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+dependencies: []
```
Comment (nmfretz) [inline alby-nostr-wallet-connect/umbrel-app.yml:29]: Since this depends on the Lightning app (LND), which in turn depends on the Bitcoin app, we can add those apps as dependencies so that a user must install Lightning and Bitcoin prior to installing Alby NWC.

```suggestion
dependencies:
  - bitcoin
  - lightning
```

They will then see this in the app store:

image

and if they try to install anyways, but are missing a dependency they will be shown this:

image

File: alby-nostr-wallet-connect/umbrel-app.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  Connect your LND node over Nostr.
+
+  Zap directly from within supported apps like Damus & Amethyst.
+
+  Amethyst step-by-step guide: https://blog.getalby.com/native-zapping-in-amethyst
+
+  Damus (currently only on Testflight) video guide: https://streamable.com/bareih
+developer: Alby
+website: http://getalby.com/
+submitter: kiwiidb
```
Comment (nmfretz) [inline alby-nostr-wallet-connect/umbrel-app.yml:19]: Would you be okay changing the submitter to `Alby`? The thought here is that we are trying to allow users to differentiate between apps submitted by the official developers of an app (you) vs apps submitted by a community member that is not related to the project.

File: alby-nostr-wallet-connect/umbrel-app.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+submission: https://github.com/getAlby/nostr-wallet-connect
+repo: https://github.com/getAlby/nostr-wallet-connect
+support: https://github.com/getAlby/nostr-wallet-connect/issues
+gallery:
+  - https://raw.githubusercontent.com/getAlby/umbrel-community-app-store/master/alby-nostr-wallet-connect/images/nwc_1.png
+  - https://raw.githubusercontent.com/getAlby/umbrel-community-app-store/master/alby-nostr-wallet-connect/images/nwc_2.png
+  - https://raw.githubusercontent.com/getAlby/umbrel-community-app-store/master/alby-nostr-wallet-connect/images/nwc_3.png
```
Comment (nmfretz) [inline alby-nostr-wallet-connect/umbrel-app.yml:26]: Thanks for these gallery images! I will upload these to https://github.com/getumbrel/umbrel-apps-gallery and now that the app will be on our app store, we can simply change this to:

```suggestion
gallery:
  - 1.png
  - 2.png
  - 3.png
```

File: alby-nostr-wallet-connect/umbrel-app.yml [hunk 7]
```diff
@@ -0,0 +1,32 @@
+manifestVersion: 1
+id: alby-nostr-wallet-connect
+name: Nostr Wallet Connect
+tagline: The power of the zaps at the tip of your fingers.
+icon: https://raw.githubusercontent.com/getAlby/umbrel-community-app-store/master/alby-nostr-wallet-connect/images/nwc_icon.svg
```
Comment (nmfretz) [inline alby-nostr-wallet-connect/umbrel-app.yml:5]: Now that this will be in our app store, we no longer need to reference the icon here

```suggestion
```

File: alby-nostr-wallet-connect/umbrel-app.yml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+version: "0.1.0"
+port: 58000
+description: >-
+  Connect your LND node over Nostr.
+
+  Zap directly from within supported apps like Damus & Amethyst.
+
+  Amethyst step-by-step guide: https://blog.getalby.com/native-zapping-in-amethyst
+
+  Damus (currently only on Testflight) video guide: https://streamable.com/bareih
```
Comment (nmfretz) [inline alby-nostr-wallet-connect/umbrel-app.yml:16]: Here's how the formatting will look in the app store:

image

If you want to, you can use 2 blank lines to create additional whitespace, and can use `-` to represent bullet points. Meaning something like this is possible:

image

File: alby-nostr-wallet-connect/umbrel-app.yml [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+developer: Alby
+website: http://getalby.com/
+submitter: kiwiidb
+submission: https://github.com/getAlby/nostr-wallet-connect
```
Comment (nmfretz) [inline alby-nostr-wallet-connect/umbrel-app.yml:20]: This can now become a reference to this PR

```suggestion
submission: https://github.com/getumbrel/umbrel-apps/pull/636
```

**Comment by nmfretz** [comment]
@kiwiidb fantastic work on this submission and the nostr wallet connect app in general! This is in great shape. I have left some suggestions in the review below that will tidy things up for integration with Umbrel.

I tested this out and had some fun sending some sats around on Amethyst ⚡️.

**Review by nmfretz — COMMENTED** [review]
@kiwiidb fantastic work on this submission and the nostr wallet connect app in general! This is in great shape. I have left some suggestions in the review below that will tidy things up for integration with Umbrel.

I tested this out and had some fun sending some sats around on Amethyst ⚡️.

**Comment by nmfretz** [comment]
While testing I ran across some minor bugs that I'll document here. These are non-blockers for going live to the app store, but hopefully these help you out.

Both of these issues are demonstrated in the screen recording below (throwaway connections):
- Error thrown in console when app connection created (no issues connecting to Amethyst as a result though).
- QR code cannot be copied to the clipboard when clicking the `Copy Secret` button.

https://github.com/getumbrel/umbrel-apps/assets/85373263/b9fd0b97-5e72-4cbc-bdee-0d9661c4cf51

On mobile screen sizes the pairing details overflow horizontally quite a bit: 
Alby - Nostr Wallet Connect (1)

File: alby-nostr-wallet-connect/umbrel-app.yml (hunk 4 above)
Comment (kiwiidb) [reply alby-nostr-wallet-connect/umbrel-app.yml:28]: Ok. If I may ask, what happened to being able to run LND with Neutrino on Umbrel? Would be nice to get up and running faster without running Bitcoin Core.

File: alby-nostr-wallet-connect/docker-compose.yml (hunk 1 above)
Comment (kiwiidb) [reply alby-nostr-wallet-connect/docker-compose.yml:11]: I've added the `data` directory, and I don't see why there would be any problems if the app is ran as a non-root user.

**Comment by kiwiidb** [comment]
Thanks @nmfretz . I've addressed all comments.
Regarding the bugs:
- The error thrown is normal and means that you have no app installed to open `nostr+walletconnect://` links. On mobile this would automatically open Amethyst, so it's normal that this does not work on desktop.
- Apparently the clipboard is unavailable on unsecured http connections: https://stackoverflow.com/questions/71873824/copy-text-to-clipboard-cannot-read-properties-of-undefined-reading-writetext . I guess we could  remove it if the connection is unsecured.
- I've fixed the typo, don't think this warrants a new release. Also, I personally don't feel the overflow on mobile is that critical, I think most people would use this app on desktop and scan the QR code with their mobile.

**Comment by nmfretz** [comment]
Sorry for the delay @kiwiidb. Thanks for addressing everything. I will test again tomorrow morning my time (PST) and then we are good to go!

Would you be okay if our designer made some slight tweaks to the gallery images/logo or would you like to keep them as-is?

Sounds good to me re the bugs.
> Apparently the clipboard is unavailable on unsecured http connections

We get around this issue using the same method as shown in the stackoverflow link you provided. See our Nostr Relay for example: https://github.com/getumbrel/umbrel-nostr-relay/blob/e4b13856b85fabe1c2e0ce789d5d959616b23196/src/components/utility/CopyText.js#L6-L22

**Comment by kiwiidb** [comment]
Yeah, you could make some tweaks, but we like our logo though :)

**Comment by nmfretz** [comment]
> Yeah, you could make some tweaks, but we like our logo though :)

Thanks @kiwiidb. Are you okay with these modifications? Our designer changed up the background a bit and updated the type to match the style of other apps in our store. Please let me know if you would rather use your originals.

**Comment by nmfretz** [comment]
And thanks again  for making all the changes. The app is working great! I made a mistake when suggesting that you had to add both bitcoin and lightning as a dependencies in the `umbrel-app.yml`. You only need to add first-degree dependencies, which would be `lightning` in your case. Sorry for the unnecessary addition. Can you please remove `bitcoin` as a dependency and let me know if you are okay with the gallery modifications, and then we go live 🐝⚡️

File: alby-nostr-wallet-connect/umbrel-app.yml [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+developer: Alby
+website: http://getalby.com/
+submitter: Alby
+submission: https://github.com/getumbrel/umbrel-apps/pull/636
+repo: https://github.com/getAlby/nostr-wallet-connect
+support: https://github.com/getAlby/nostr-wallet-connect/issues
+gallery:
+  - 1.png
+  - 2.png
+  - 3.png
```
Comment (nmfretz) [inline alby-nostr-wallet-connect/umbrel-app.yml:26]:
```suggestion
gallery:
  - 1.jpg
  - 2.jpg
  - 3.jpg
```

**Review by nmfretz — COMMENTED** [review]
Excellent, thanks @kiwiidb! I have tested the new changes on amd64 and arm64 and everything is working well. 

I just realized that with the new gallery images, we need to change the `.png`'s to `.jpg`'s in the umbrel-apps.yml. See below for suggested change.

Sorry for the last minute delay. I tried to commit a change to your umbrel-apps repo to avoid having to ask you to do it, but I don't have write permission.

So close!

**Comment by nmfretz** [comment]
@kiwiidb pinging again in case you missed this. If you give me edit access to the repo I can handle this for you no problem.

**Comment by kiwiidb** [comment]
Done

**Comment by nmfretz** [comment]
Thanks @kiwiidb, going live to the app store now! Thanks for all your work bringing this to Umbrel, we really appreciate it. What a fantastic addition.
